src/onegov/org/views/payment.py

```diff
@@ -102,36 +108,138 @@ def send_ticket_notifications(
+                        ticket = get_ticket_for_link(link)
+
+                if ticket and ticket.id not in seen_ticket_ids:
+                    seen_ticket_ids.add(ticket.id)
+                    yield ticket
+    if not payments:
+        request.warning(_('No payments found for PDF generation'))
+        return request.redirect(request.class_link(PaymentCollection))
+
+    filename = 'Payments.pdf'
+    deduplicated_tickets = set(get_tickets_for_pdf(payments))
+    multi_pdf = TicketsPdf.from_tickets(request, deduplicated_tickets)
+    return Response(
+        multi_pdf.read(),
+        content_type='application/pdf',
+        content_disposition=f'inline; filename={filename}'
+    )
+
+
+@OrgApp.form(
     model=PaymentCollection,
     template='payments.pt',
+    form=PaymentSearchForm,
     permission=Private
 )
 def view_payments(
     self: PaymentCollection,
     request: OrgRequest,
+    form: PaymentSearchForm,
     layout: PaymentCollectionLayout | None = None
-) -> RenderData:
+) -> RenderData | WebobResponse:
+    request.include('invoicing')
+    layout = layout or PaymentCollectionLayout(self, request)
 
-    tickets = TicketCollection(request.session)
+    if form.submitted(request):
+        form.update_model(self)
+        return request.redirect(request.link(self))
+
+    if not form.errors:
+        form.apply_model(self)
+
+    if request.params.get('format') == 'pdf':
+        return handle_pdf_response(self, request)
 
     providers = {
         provider.id: provider
         for provider in PaymentProviderCollection(request.session).query()
     }
 
-    payment_links = self.payment_links_by_batch()
-
     return {
-        'title': _('Payments'),
+        'title': _('Receivables'),
+        'form': form,
         'layout': layout or PaymentCollectionLayout(self, request),
         'payments': self.batch,
-        'get_ticket': partial(ticket_by_link, tickets),
+        'tickets': self.tickets_by_batch(),
+        'reservation_dates': self.reservation_dates_by_batch(),
         'providers': providers,
-        'payment_links': payment_links
+        'payment_links': self.payment_links_by_batch(),
+        'pdf_export_link': append_query_param(request.url, 'format', 'pdf')
     }
 
 
+@OrgApp.json(
+    model=PaymentCollection,
+    name='batch-set-payment-state',
+    request_method='POST',
+    permission=Private
+)
+def handle_batch_set_payment_state(
+    self: PaymentCollection,
+    request: OrgRequest
+) -> JSON_ro:
+
+    request.assert_valid_csrf_token()
+    payment_ids = request.json_body.get('payment_ids', [])
+    state = request.json_body.get('state')
+
+    if state not in ('invoiced', 'paid', 'open'):
+        raise exc.HTTPBadRequest()
+
+    payments_query = self.session.query(Payment).distinct().filter(
+        Payment.id.in_(payment_ids)
+    )
+    updated_count = 0
+    for payment in payments_query:
+        if payment.state != state:
+            payment.state = state
```

Comment:
This seems very dangerous. Marking as `paid`/`open` should only be allowed for manual payments. `invoiced` I'm less sure about, if the workflow is `paid` -> `invoiced`, then it should only be possible on `paid` (manual?) payments. If it's `open` -> `invoiced` -> `paid` then it should only be possible on `open` manual payments.